lib/Mojolicious/Plugin/Config.pm

```diff
@@ -9,9 +9,25 @@ sub load { $_[0]->parse(decode('UTF-8', path($_[1])->slurp), @_[1, 2, 3]) }
 sub parse {
   my ($self, $content, $file, $conf, $app) = @_;
 
+  # add the "#line" directive *only* if it's safe to do so
+  my $line_directive = $file =~ /[\r\n]/
+    ? ''    # can't break "#line" across lines
+    : (
+    $file =~ /\s/
+    ? (
+      $file =~ /"/
+      ? ''    # can't escape double-quotes
+      : qq{\n#line 1 "$file"\n}
+      )
+    : qq{\n#line 1 $file\n}
+    );
```

Comment:
This is really ugly code.

Reply:
It's the same code as before, but run through perltidy, as per the Rules in Mojolicious::Guides::Contributing. I could rewrite it as `if` statements for example - what is your preference?

Reply:
I don't mean formatting alone. The code also creates lots of untested special cases. Unless the whole feature can be reduced to say one line, i doubt i would vote for it.

Reply:
Ah, I see, and yes, [I agree](https://github.com/mojolicious/mojo/pull/1494#issuecomment-615942450). I've made some updates to attempt to address it, [see my comment below](https://github.com/mojolicious/mojo/pull/1494#issuecomment-623005604).